src/ast/rewriter/array_rewriter.cpp

```diff
@@ -198,13 +198,97 @@ bool array_rewriter::squash_store(unsigned n, expr* const* args, expr_ref& resul
 
 br_status array_rewriter::mk_select_core(unsigned num_args, expr * const * args, expr_ref & result) {
     SASSERT(num_args >= 2);
+    expr *arg0 = args[0];
+    expr_ref tmp(m());
+    bool first = true;
+
+#define RET(x, status)            \
```

Comment:
why not a separate function that returns a status? BR_FAILED results in trying the next rewrite.
The local variable arg0 is updated in the first part of the code and should not be used in the second part because it could be unrelated to args

Reply:
Sorry, I didn't understand the suggestion.
arg0 is only used in this new loop; it's not used in the old code.